### PR TITLE
Feature/initial serializer

### LINKED ISSUE #1: Serialize an object to json

### LINKED ISSUE #2: Deserialize a json object back to an object

### LINKED ISSUE #4: Fix the autoloading issue in tests in the feature\

### PR DESCRIPTION
**Pull request by brettminnie**
Closes #1 and closes #2 
Fixes #4

### CODE REVIEW

File: .scrutinizer.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    php_code_sniffer: true
+    sensiolabs_security_checker: true
+    php_code_coverage: true
+    php_pdepend: true
+    php_loc:
+        enabled: true
+        excluded_dirs: [vendor]
+    php_cpd:
+        enabled: true
+        excluded_dirs: [vendor]
```
Comment (eddiejaoude) [inline .scrutinizer.yml]: Extra blank line to remove the red sign from github

**Comment by eddiejaoude** [comment]
:+1:

File: .scrutinizer.yml (hunk 1 above)
Comment (brettminnie) [inline .scrutinizer.yml]: Done

**Comment by brettminnie** [comment]
Ok, added the missing new line, merge if you are happy, or just clap your hands

**Pull request merged**